addons/website_sale_collect/static/src/js/payment_form.js

```diff
@@ -0,0 +1,41 @@
+import { _t } from '@web/core/l10n/translation';
+import PaymentForm from '@payment/js/payment_form';
+
+PaymentForm.include({
+    /**
+     * Keep payment button context to switch labels displayed on payment method change.
+     *
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+        const buttons = document.querySelectorAll('button[name="o_payment_submit_button"]');
+        this.paymentButtonContext = {
+            buttonLabelMapping: [...buttons].map((btn) => [btn, btn.textContent]),
+            payOnSiteLabel: _t("Confirm"),
+        };
```

Comment:
As was discussed with @AntoineVDV, even if we have multiple payment buttons on the page the idea that they should still be the same (for mobile and desktop), so we could simplify the fix to just store one default label.
```suggestion
        this.submitButtonDefaultLabel = this.buttons[0]?.textContent;
```